Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -181,11 +180,18 @@ BTDiagnostics::DoDump (int step, int i_buffer, bool force_flush)
 
 
 bool
-BTDiagnostics::DoComputeAndPack (int step, bool /*force_flush*/)
+BTDiagnostics::DoComputeAndPack (int step, bool force_flush)
 {
     // always set to true for BTDiagnostics since back-transform buffers are potentially
-    // computed and packed every timstep, except at initialization when step == -1.
-    return (step>=0);
+    // computed and packed every timstep, except at initialization when step == -1, or when
+    // force_flush is set to true.
```

Comment:
```suggestion
    // force_flush is set to true, because we dont need to redundantly re-compute buffers when force_flush = true. We only need to dump the buffers. Note that the BTD computation is performed every timestep (step>=0)
```